tensorflow/python/ops/variable_scope.py

```diff
@@ -683,6 +683,8 @@ def _get_single_variable(self,
         init_val = initializer
         variable_dtype = None
       else:
+        if type(initializer)==type:
```

Comment:
I don't understand what this line does, nor why it's needed. Please add an extensive comment explaining what's happening here.